### PR TITLE
docs: modernize backend test guidance

### PR DESCRIPTION
**Pull request by andife**
## Draft status

This PR is intentionally a **Draft** and should not be merged yet. It provides a concrete documentation proposal for two linked discussions:

- #8287 - clarify the normative status and lifecycle of backend tests
- #8288 - define supported access to serialized backend node-test fixtures after #7959

The proposed specification-status wording is visibly marked as Draft text in `OnnxBackendTest.md`. It should be confirmed or revised by the Operators SIG before this PR is marked ready for review.

## Motivation

The backend guides combine statements written in 2017 with an implementation that has changed substantially:

- representative model tests switched from remote heavy models to local lightweight fixtures in #4861;
- generated node `.onnx` and `.pb` artifacts were removed in #7959 and cases are now constructed in memory;
- downstream projects had consumed the previously documented file layout;
- the implementation examples point to archived or unmaintained integrations;
- the coverage example hard-codes the historical total of 70 operators;
- the current text does not clearly distinguish the normative specification from executable conformance cases.

## Proposed changes

- describe backend tests as conformance cases whose coverage is limited to the loaded cases
- add proposed wording for their relationship to the IR and versioned operator specifications
- document the current node, simple-model, lightweight real-model, and retained converted-model categories
- document the in-memory node-test API and the compatibility discussion for file-based consumers
- replace historical Caffe2/CoreML/TensorFlow examples with the in-repository reference backend and maintained ONNX-TensorRT integration
- replace the static coverage snapshot and operator count with the current dynamic report semantics
- remove the obsolete cloud-upload/admin contribution instructions

## Questions for reviewers

1. Does #8287's proposed normative/informative/conformance distinction match the Operators SIG's intent?
2. Which loader fields and serialized layouts should be documented as stable interfaces under #8288?
3. Should the factual cleanup be split and merged before the policy wording is resolved?
4. Is ONNX-TensorRT the preferred maintained external example, or should the guide use only the in-repository reference backend?

## Validation

- `git diff --check`
- verified every local Markdown target in the two changed files exists
- checked the documented categories and in-memory access path against the current loader and runner
- checked the integration example against the in-repository reference backend and current ONNX-TensorRT backend test

`pixi run lint` could not complete because the environment failed to download conda-forge packages with TLS `UnknownIssuer`. No lint failure from the changed Markdown files was reported before dependency resolution stopped.

### CODE REVIEW

File: docs/OnnxBackendTest.md [hunk 1]
```diff
@@ -8,18 +8,86 @@ SPDX-License-Identifier: Apache-2.0
...
-- **Model Tests** verify the backend at the model level. The test cases are similar to those of Node Tests', but instead of a node, the backend will be given an ONNX model.
+> **Draft note:** The precise normative status and lifecycle of backend tests is being
+> discussed in [issue #8287](https://github.com/onnx/onnx/issues/8287). This
+> section is proposed wording for that discussion and will be finalized before
+> this documentation change is marked ready for review.
+
+The ONNX IR and versioned operator specifications define normative ONNX
+semantics. Backend tests provide executable conformance cases and examples
+derived from those specifications. They are also useful evidence when
+clarifying an ambiguity, but a test does not silently override a versioned
```
Comment (cbourjau) [inline docs/OnnxBackendTest.md]: You state so below, but I'd not put this sentence here at all. If the standard is ambiguous, we should _not_ look at any of these tests for guidance. If the standard was written ambiguously, it is just as likely that the test were  written obliviously of a particular corner case. We must fix such ambiguities conciously and with high priority rather than letting a random test decide.

File: docs/OnnxBackendTest.md [hunk 2]
```diff
@@ -8,18 +8,86 @@ SPDX-License-Identifier: Apache-2.0
...
+- **Real model tests** exercise representative models. The current models are
+  lightweight fixtures committed under
+  [`onnx/backend/test/data/light`](/onnx/backend/test/data/light), with metadata
+  under [`onnx/backend/test/data/real`](/onnx/backend/test/data/real).
+- **PyTorch-converted and PyTorch-operator tests** are historical model fixtures
+  retained under [`onnx/backend/test/data`](/onnx/backend/test/data).
+
+### Node-test data access
+
+PR [#7959](https://github.com/onnx/onnx/pull/7959) removed the generated node
```
Comment (cbourjau) [inline docs/OnnxBackendTest.md]: Let's not put a tombstone comment about how things were. Let's document what is today.

File: docs/OnnxBackendTest.md [hunk 3]
```diff
@@ -8,18 +8,86 @@ SPDX-License-Identifier: Apache-2.0
...
+`.onnx` and `.pb` artifacts from the package. Node cases are now constructed in
+memory. Consumers should load them through
+[`load_model_tests(kind="node")`](/onnx/backend/test/loader/__init__.py) and use
+each returned `TestCase.model` and `TestCase.data_sets` instead of assuming that
+`onnx/backend/test/data/node` exists.
+
+Earlier documentation described the removed directory as the source of truth,
```
Comment (cbourjau) [inline docs/OnnxBackendTest.md]: That is a bit out of context and also was only use once in passing, afaik. I don't think this was ever ment with respect to being the source of truth compaired to the python code that produced it. I read it to just mean that downstream runtimes can find the expected output in those files.
